java/daikon/dcomp/BuildJDK.java

```diff
@@ -3,375 +3,500 @@
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.jar.JarEntry;
 import java.util.jar.JarFile;
-import java.util.zip.ZipException;
 import org.apache.bcel.*;
 import org.apache.bcel.classfile.ClassParser;
 import org.apache.bcel.classfile.JavaClass;
 import org.apache.bcel.generic.*;
 
 /**
- * Converts each file in the JDK. Each method is doubled. The new methods are distinguished by a
- * final parameter of type ?? and are instrumented to track comparability. User code will call the
- * new methods, but instrumentation code is unchanged and calls the original methods.
+ * BuildJDK uses {@link DCInstrument} to add comparability instrumentation to a set of Java class
+ * files and then stores the modified files into a directory identified by a (required) command line
+ * argument.
+ *
+ * <p>DCInstrument duplicates each method of a class file. The new methods are distinguished by the
+ * addition of a final parameter of type DCompMarker and are instrumented to track comparability.
+ * Based on its invocation arguments, DynComp will decide whether to call the instrumented or
+ * uninstrumented version of a method.
  */
 public class BuildJDK {
 
-  @Option(
-      "Instrument the given classfiles from the specified source directory (by default, src must be a jar file)")
-  public static boolean classfiles = false;
-
-  /** Synopsis for the dcomp command line. */
-  public static final String synopsis =
-      "daikon.BuildJDK [options] src dest [class_prefix | classfiles...]";
-
   /**
-   * Given an explicit {@code rt.jar} filename, or a root JDK or JRE directory, finds the {@code
-   * rt.jar} file.
+   * The "java.home" system property. Note that there is also a JAVA_HOME variable that contains
+   * {@code System.getenv("JAVA_HOME")}.
    */
-  public static class RtJarFinder {
-    private final String arg;
-
-    public RtJarFinder(String arg) {
-      super();
-      this.arg = arg;
-    }
+  public static final String java_home = System.getProperty("java.home");
 
-    /**
-     * @param spaceSeparatedJarLocations a list of potential jar filenames to look for, in case arg
-     *     was a directory. For example, if arg was /usr/bin/jd2sdk, and spaceSeparatedJarLocations
-     *     was "jre/lib/rt.jar", this could return /usr/bin/jd2sdk/jre/lib/rt.jar, if it existed.
-     */
-    public String findRtJar(String spaceSeparatedJarLocations) {
-      String jarFilename = arg;
-      if (!arg.endsWith("jar")) {
-        for (String candidate : spaceSeparatedJarLocations.split(" ")) {
-          String rt = jarFilename + candidate;
-          if (exists(rt)) {
-            return rt;
-          }
-        }
-      }
-      return jarFilename;
-    }
+  /**
+   * Instrument the class files given on the command line instead of the ones in the Java runtime
+   * library.
+   */
+  @Option("Instrument only the class files given on the command line")
+  public static boolean classfiles = false;
```

Comment:
I don't think this command-line argument is necessary.  It can be inferred from the presence of command-line arguments.  In particular:  if there is one command-line argument, then instrument the JDK, and if there are more, instrument the given classes.  Little logic and documentation need to be changed other than removing this command-line argument.

Reply:
Simplified code and looks good - thanks for suggestion.